utilities/cli/holohub.py

```diff
@@ -1025,6 +1062,79 @@ def handle_clear_cache(self, args: argparse.Namespace) -> None:
+        """Handle create command"""
+        try:
+            import cookiecutter.main
+        except ImportError:
+            self._install_template_deps(args.dryrun)
+
+        import cookiecutter.exceptions
+        import cookiecutter.main
+
+        # Ensure template directory exists
+        template_dir = self.HOLOHUB_ROOT / args.template
+        if not template_dir.exists():
+            holohub_cli_util.fatal(f"Template directory {template_dir} does not exist")
+
+        canonical_project_name = args.project.lower().replace(" ", "_")
+        project_dir = self.HOLOHUB_ROOT / "applications" / canonical_project_name
+        context = {
+            "project_name": args.project,
+            "project_slug": canonical_project_name,
+            "full_name": os.getenv("USER", "HoloHub User"),
+            "affiliation": "Holoscan",
+            "language": args.language,
+            "version": "0.1.0",
+            "holoscan_version": "2.3",
+            "platforms": '["x86_64", "aarch64"]',
+            "tags": '["keyword1", "keyword2", "keyword3"]',
+            "description": "A brief description of the project",
+            "license": "Apache-2.0",
+            "year": datetime.datetime.now().year,
+        }
+
+        # Print summary if dryrun
+        if args.dryrun:
+            print("Would create project with these parameters (dryrun):")
+            print(f"Directory: {project_dir}")
+            for key, value in context.items():
+                print(f"  {key}: {value}")
+            return
+
+        if project_dir.exists():
+            holohub_cli_util.fatal(f"Project directory {project_dir} already exists")
+
+        try:
+            cookiecutter.main.cookiecutter(
+                str(template_dir),
+                no_input=True,
+                extra_context=context,
+                output_dir=str(self.HOLOHUB_ROOT / "applications"),
+            )
+        except Exception as e:
+            holohub_cli_util.fatal(f"Failed to create project: {str(e)}")
+
+        # Copy the appropriate example source file
+        template_file = template_dir / f"example.{args.language}"
+        if template_file.exists():
+            src_dir = project_dir / "src"
+            src_dir.mkdir(exist_ok=True)
+            target_file = src_dir / f"{canonical_project_name}.{args.language}"
+            shutil.copy2(str(template_file), str(target_file))
+
+        print(
+            Color.green(f"Successfully created new project: {args.project}"),
+            f"\nProject directory: {project_dir}\n\n"
+            f"Possible next steps:\n"
+            f"- Add your operators to {target_file}\n"
+            f"- Update project metadata in {project_dir / 'metadata.json'}\n"
+            f"- Review source code license files and headers (e.g. {project_dir / 'LICENSE'})\n"
+            f"- Build and run your application:\n"
+            f"   ./holohub build {canonical_project_name}\n"
```

Comment:
```suggestion
```

Reply:
Hi @tbirdso, after `./holohub create my_proj`, it seems I still need to add `add_holohub_application(my_proj)` to `applications/CMakeLists.txt`.

 otherwise `./holohub run my_poj` won't work. is there a way to avoid modifying `applications/CMakeLists.txt`?

Reply:
(@jjomier mentioned that this could be a new feature to support `./holohub run my_proj` when `my_proj` is not in the standard directory.)

Reply:
Is it possible to use `cookiecutter` or otherwise append to `applications/CMakeLists.txt` when `holohub create` is run? You're right, at the moment it is required that C++ projects (or any project leveraging the CMake build system) have an `add_holohub_application` entry.

Reply:
added a function to modify the cmakelists.txt https://github.com/wyli/holohub/blob/2f7745f98b198c6c37a9d0dcfdf6e7ec2095cf58/utilities/cli/holohub.py#L1078-L1079 I think we can further improve 'build' and 'run' to process folders with local cmakelists.txt.